src/smart_markers/mod_smart_markers_rdbms_async.erl

```diff
@@ -0,0 +1,118 @@
+%%%----------------------------------------------------------------------------
+%%% @copyright (C) 2020, Erlang Solutions Ltd.
+%%% @doc
+%%%    RDBMS backend for mod_smart_markers
+%%% @end
+%%%----------------------------------------------------------------------------
+-module(mod_smart_markers_rdbms_async).
+-behavior(mod_smart_markers_backend).
+-behaviour(mongoose_aggregator_worker).
+
+-include("jlib.hrl").
+-include("mongoose_logger.hrl").
+
+-export([init/2, stop/1, update_chat_marker/2, get_chat_markers/4, get_conv_chat_marker/6]).
+-export([remove_domain/2, remove_user/2, remove_to/2, remove_to_for_user/3]).
+
+%% Worker callbacks
+-export([request/2, aggregate/3, verify/3]).
+
+%%--------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------
+-spec init(mongooseim:host_type(), gen_mod:module_opts()) -> ok.
+init(HostType, Opts) ->
+    AsyncOpts = prepare_pool_opts(Opts),
+    mod_smart_markers_rdbms:init(HostType, Opts),
+    start_pool(HostType, AsyncOpts),
+    ok.
+
+stop(HostType) ->
+    mongoose_async_pools:stop_pool(HostType, smart_markers).
+
+prepare_pool_opts(#{async_writer := AsyncOpts}) ->
+    AsyncOpts#{pool_type => aggregate,
+               request_callback => fun ?MODULE:request/2,
+               aggregate_callback => fun ?MODULE:aggregate/3,
+               verify_callback => fun ?MODULE:verify/3}.
+
+-spec start_pool(mongooseim:host_type(), mongoose_async_pools:pool_opts()) -> term().
+start_pool(HostType, Opts) ->
+    catch mongoose_async_pools:start_pool(HostType, smart_markers, Opts).
```

Comment:
report errors?

Reply:
This only happens during testing, where pools are turned on and off. For some reason out of the scope of this PR ejabberd_sup was still having the children somewhere there.